### PR TITLE
chore: fix type errors caused by canary

### PR DESCRIPTION
**Pull request by kt3k**
Some type errors started appearing because of TS upgrade in canary (See https://github.com/denoland/deno_std/actions/runs/5450255896/jobs/9915333243 for details). This PR fixes those type errors.

### CODE REVIEW

**Review by dsherret — APPROVED** [review]
Interesting… seems worse at figuring stuff out.

File: http/file_server_test.ts [hunk 1]
```diff
@@ -130,7 +130,7 @@ async function fetchExactPath(
   const encoder = new TextEncoder();
   const decoder = new TextDecoder();
   const request = encoder.encode("GET " + path + " HTTP/1.1\r\n\r\n");
-  let conn: void | Deno.Conn;
+  let conn: void | Deno.Conn = undefined;
```
Comment (aapoalas) [inline http/file_server_test.ts:133]: The `void` here is strange. I'd definitely use `undefined` here instead. Would it work without assignment with that?

File: http/server_sent_event.ts [hunk 2]
```diff
@@ -345,10 +345,11 @@ export class ServerSentEventStreamTarget extends EventTarget
    * {@linkcode ResponseInit} needed to create a response that will establish
    * a SSE connection with the client. */
   asResponseInit(responseInit: ResponseInit = {}): [BodyInit, ResponseInit] {
-    responseInit.headers = new Headers(responseInit.headers);
+    const headers = new Headers(responseInit.headers);
```
Comment (aapoalas) [inline http/server_sent_event.ts:348]: This ends up being a fairly okay perf boost as well, nice! :)

**Review by aapoalas — APPROVED** [review]
Looks good to me, though I'd personally change one thing.

File: http/file_server_test.ts (hunk 1 above)
Comment (kt3k) [reply http/file_server_test.ts:133]: Oh, that seems working. Thanks!

**Comment by aapoalas** [comment]
> Interesting… seems worse at figuring stuff out.

At work my friend was saying that basic if-else type clauses have stopped working in some cases in newer TS. There may be some sort of regression happening.

**Pull request merged**